### ISSUE TITLE
Disallow calling `configure` inside of a test

### ISSUE THREAD

**Issue by Janpot** [testing-library/eslint-plugin-testing-library#997]
### Name for new rule

prefer-top-level-configure

### Description of the new rule

Fail when `configure` is not called top-level

### Testing Library feature

global [configuration](https://testing-library.com/docs/dom-testing-library/api-configuration/) utility

### Testing Library framework(s)

`@testing-library/dom`, `@testing-library/react`, perhaps others

### What category of rule is this?

Warns about a potential error

### Code examples

```tsx
import { configure } from '@testing-library/react'
test('some test', () => {
  // FAIL:
  configure({ reactStrictMode: true })

  // ...
})
```

```tsx
import { configure } from '@testing-library/react'

// OK:
configure({ reactStrictMode: true })

test('some test', () => {
  // ...
})
```

### Do you want to submit a pull request to make the new rule?

Yes

**Comment by Belco90** [comment]
Hi @Janpot. I'm checking the docs for the `configure` util and I don't see any reference in React or other Testing Library frameworks, so I believe this would only apply to DOM.

**Comment by Janpot** [comment]
https://testing-library.com/docs/react-testing-library/api#configure

can't really speak for other libs, but searching on the docs for "configure" suggests other libs have this as well.

**Comment by Belco90** [comment]
@Janpot Good catch! I couldn't find anything under the Setup section.
